### PR TITLE
Fourth task commit. Waiting for review.

### PR DESCRIPTION
**Pull request by Radu-Stroe**

### CODE REVIEW

File: src/test/java/internship/tests/MyNewslettersTest.java [hunk 1]
```diff
@@ -12,22 +12,42 @@
...
+        login();
+        MyAccountPage myAccountPage = new MyAccountPage().clickMyNewslettersButton()
+                                                         .navigateMyAccount()
+                                                         .addNewsletter();
+        softAssert.get().assertTrue(myAccountPage.isValidateNewsletterDisplayed(),
+                "Validate newsletter element is not displayed.");
+        softAssert.get().assertTrue(myAccountPage.checkValidateNewsletterText(),
+                "Validating text from adding newsletter has failed.");
         softAssert.get().assertAll();
     }
```
Comment (cristinanedin) [inline src/test/java/internship/tests/MyNewslettersTest.java:54]: remove ThreadLocal<SoftAssert> in tear down

File: src/test/java/internship/framework/pages/MyAccountPage.java [hunk 2]
```diff
@@ -11,9 +11,45 @@
...
+    {
+        ProjectLogger.info("Check if validate newsletter element is displayed.");
+        return ElementUtils.doesAppear(validateNewsletter);
+
+    }
+
+    public boolean checkValidateNewsletterText()
+    {
+        ProjectLogger.info("Check if validate newsletter text matches.");
+        return ElementUtils.getElementText(validateNewsletter).contains("We've added");
```
Comment (cristinanedin) [inline src/test/java/internship/framework/pages/MyAccountPage.java]: make this method String. Add the check in test.

**Review by cristinanedin — CHANGES_REQUESTED** [review]

File: src/test/java/internship/framework/pages/MyAccountPage.java (hunk 2 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/pages/MyAccountPage.java]: Fixed.

File: src/test/java/internship/framework/pages/MyAccountPage.java [hunk 3]
```diff
@@ -40,10 +40,10 @@ public boolean isValidateNewsletterDisplayed()
 
     }
 
-    public boolean checkValidateNewsletterText()
+    public String checkValidateNewsletterText()
```
Comment (cristinanedin) [inline src/test/java/internship/framework/pages/MyAccountPage.java]: getValidateNewsletterText() would be a better naming

File: src/test/java/internship/tests/MyNewslettersTest.java [hunk 4]
```diff
@@ -46,7 +46,7 @@ public void TestingMyAccountFunctionality()
                                                          .addNewsletter();
         softAssert.get().assertTrue(myAccountPage.isValidateNewsletterDisplayed(),
                 "Validate newsletter element is not displayed.");
-        softAssert.get().assertTrue(myAccountPage.checkValidateNewsletterText(),
+        softAssert.get().assertTrue(myAccountPage.checkValidateNewsletterText().contains("We've added"),
```
Comment (cristinanedin) [inline src/test/java/internship/tests/MyNewslettersTest.java]: please use an object for "We've added"

File: src/test/java/internship/framework/pages/MyAccountPage.java (hunk 3 above)
Comment (Radu-Stroe) [reply src/test/java/internship/framework/pages/MyAccountPage.java]: Noted and fixed.

File: src/test/java/internship/tests/MyNewslettersTest.java (hunk 4 above)
Comment (Radu-Stroe) [reply src/test/java/internship/tests/MyNewslettersTest.java]: Noted and fixed.

**Review by cristinanedin — APPROVED** [review]

**Pull request merged**